lib/membrane/rtp/ssrc_router.ex

```diff
@@ -157,22 +162,30 @@ defmodule Membrane.RTP.SSRCRouter do
       state =
         state
         |> put_in([:input_pads, ssrc], pad)
-        |> put_in([:linking_buffers, ssrc], [])
+        |> put_in([:buffered_actions, ssrc], [])
 
       {[notify: {:new_rtp_stream, ssrc, payload_type, extensions}], state}
     end
   end
 
-  defp maybe_add_to_linking_buffer(type, value, ssrc, state) do
+  defp maybe_buffer_action(:end_of_stream, ssrc, ctx, state) do
+    action = {:end_of_stream, Pad.ref(:output, ssrc)}
+    do_maybe_buffer_action(action, ssrc, ctx, state)
+  end
+
+  defp maybe_buffer_action(type, value, ssrc, ctx, state) do
     action = {type, {Pad.ref(:output, ssrc), value}}
+    do_maybe_buffer_action(action, ssrc, ctx, state)
+  end
```

Comment:
Hmm I am not sure if we should distinguish between different `Membrane.Element.Action.t` types in function signature...I know it is just a private function, but personally I find it less readable. How about removing these two "converter" functions (`maybe_buffer_action/4` and maybe_buffer_action/5`) and explicitly calling `do_maybe_buffer_actions/4` (obviously renamed to `maybe_buffer_actions/4` with the `Membrane.Element.Action.t` as first argument?